tutorial.md

```diff
@@ -0,0 +1,231 @@
+CREATE TABLE ks.t (pk int, ck int, v int, vs set<int>, PRIMARY KEY (pk, ck)) WITH cdc = {'enabled': 'true'}
+```
+
+## Creating the CDC consumer
+The most important part of using the library is to define a callback that will be executed 
+after reading a CDC log from the database. 
+Such callback is defined by implementing the `Consumer` trait located in `scylla-cdc::consumer`. 
+As for now, we will define a struct with no member variables for this purpose:
+```rust
+struct TutorialConsumer;
+```
+Since the callback will be executed asynchronously, 
+we have to use the [async-trait](https://crates.io/crates/async-trait) crate 
+to implement the `Consumer` trait. 
+We also use the [anyhow](https://crates.io/crates/anyhow) crate for error handling.
+```rust
+use anyhow;
+use async_trait::async_trait;
+use scylla_cdc::consumer::*;
+
+struct TutorialConsumer;
+
+#[async_trait]
+impl Consumer for TutorialConsumer {
+    async fn consume_cdc(&mut self, _data: CDCRow<'_>) -> anyhow::Result<()> {
+        println!("Hello, scylla-cdc!");
+        Ok(())
+    }
+}
+```
+
+We'll cover on how to use the `CDCRow` structure in [Using CDCRow](tutorial.md#using-cdcrow).
+
+The library is going to create one instance of `TutorialConsumer` per CDC stream, so we also need to define
+a `ConsumerFactory` for them:
+```rust
+struct TutorialConsumerFactory;
+
+#[async_trait]
+impl ConsumerFactory for TutorialConsumerFactory {
+    async fn new_consumer(&self) -> Box<dyn Consumer> {
+        Box::new(TutorialConsumer)
+    }
+}
+```
+
+### Adding shared state to the consumer
+Different instances of `Consumer` are being used in separate Tokio tasks. 
+Due to that, the runtime might schedule them on separate threads.
+
+Because of that, a struct implementing the `Consumer` trait should also implement `Send` trait 
+and a struct implementing the `ConsumerFactory` trait should implement `Send` and `Sync` traits.
+Luckily, Rust implements these traits by default if all member variables of a struct implement them.
+
+If the consumers need to share some state, like a reference to an object, 
+they can be wrapped in an [Arc](<https://doc.rust-lang.org/std/sync/struct.Arc.html>).
+
+An example of that might be a `Consumer` that counts rows read by all its instances:
+
+```rust
+use atomic_counter::{AtomicCounter, ConsistentCounter};
+use tokio::sync::Mutex;
+
+struct CountingConsumer {
+    counter: Arc<Mutex<ConsistentCounter>>
+}
+
+#[async_trait]
+impl Consumer for CountingConsumer {
+    async fn consume_cdc(&mut self, _: CDCRow<'_>) -> anyhow::Result<()> {
+        let mut counter = self.counter.lock().await;
+        counter.add(1);
+        println!("Row no.{}", counter.get());
+        Ok(())
+    }
+}
+```
+(an external crate [atomic_counter](https://docs.rs/atomic-counter/latest/atomic_counter/) was used in this example)
+
+<b>Note:</b> in general, keeping a mutable state in the `Consumer` is not recommended, 
+since it requires synchronization (i.e. a mutex),
```

Comment:
I'm guessing that you added a mutex to the `CountingConsumer` so that this statement makes more sense? Atomics are more costly than non-atomic integers, especially if they are accessed from multiple threads at once, so the advice applies to them, too.